staging/src/k8s.io/client-go/tools/record/events_cache_test.go

```diff
@@ -278,3 +278,105 @@ func TestEventCorrelator(t *testing.T) {
 		}
 	}
 }
+
+func TestEventSpamFilter(t *testing.T) {
+	spamKeyFuncBasedOnObjectsAndReason := func(e *v1.Event) string {
+		return strings.Join([]string{
+			e.Source.Component,
+			e.Source.Host,
+			e.InvolvedObject.Kind,
+			e.InvolvedObject.Namespace,
+			e.InvolvedObject.Name,
+			string(e.InvolvedObject.UID),
+			e.InvolvedObject.APIVersion,
+			e.Reason,
+		},
+			"")
+	}
+	firstEvent := makeEvent("first", "i am first", makeObjectReference("Pod", "my-pod", "my-ns"))
+	duplicateEvent := makeEvent("duplicate", "me again", makeObjectReference("Pod", "my-pod", "my-ns"))
+	similarEvent := makeEvent("first", "me again", makeObjectReference("Pod", "my-pod", "my-ns"))
+	testCases := map[string]struct {
+		previousEvents  []v1.Event
```

Comment:
Nit, looks like previous event is always the same, it might be more readable to not include `previousEvents` field in the test case

Reply:
Removed previousEvents field